### ISSUE TITLE
rfsoc_sam for ZCU216 kit

### ISSUE THREAD

**Issue by KAUSHAL1986** [strath-sdr/rfsoc_sam#12]
I want to compile this project completely for ZCU216 kit. Can anybody list the detailed instructions to do this ??

**Comment by KAUSHAL1986** [comment]
This is where I have reached ,please advice further....
1. The project is created for ZCU216 kit. I am attaching the .xsa zip file for your reference which contains hardware handoff file.
2. the difference between ZCU111 and ZCU216 kit w.r.t CLK104 card is the gpio pins from PL side extended for programming of 1 LMK and 2 LMX devices placed on the CLK104 card. This I achieved by modifying the drivers for rfclk and selecting GPIO lines from PYNQ after PL is loaded with bitstream generated from point 1.
3. But now i am stuck with the following error (see the attached pynq log file). this error comes after running the following commands in the order mentioned below :
 
from overlay import Overlay
sam = Overlay()
analyzer = sam.spectrum_analyzer()
4. the .xsa zipfiles have been renamed to .zip file for attaching here as .xsa attachments are not supported.
5. 
please advice...

With thanks and Regards
KAUSHAL KUMAR
[pynq_log.txt](https://github.com/strath-sdr/rfsoc_sam/files/7459465/pynq_log.txt)
[block_design_top.zip](https://github.com/strath-sdr/rfsoc_sam/files/7459477/block_design_top.zip)

**Comment by dnorthcote** [comment]
Hi Kaushal,

Are you able to provide me with the .tcl file for the IP Integrator board design? This will allow me to regenerate your project on my end.

I believe the issue you are experiencing is due to the configuration of the clocks on the board. The LMX feeds into a particular DAC and ADC on the Gen3 RFSoC device. Then this clock is distributed to the other tiles. Additionally, there are also external connections on the CLK104 board that need to be created.

I can verify a few things for you to try if you pass me the .tcl file. I will be able to provide you more information this evening if we will support the ZCU216,

Kind regards,
David.

**Comment by KAUSHAL1986** [comment]
please find attached the tcl file for the whole project(please make necessary changes for ip repo paths and all, I am novice with tcl please bear with me...so I am attaching all the tcl scripts in project path
![rfdc1](https://user-images.githubusercontent.com/42259493/139838881-c99e4c4c-449e-465d-9f1d-ac64a98439bb.png)
![rfdc2](https://user-images.githubusercontent.com/42259493/139838891-cc707a76-55d8-4992-b0f7-12ffc35c6c92.png)
[tcl_scripts.zip](https://github.com/strath-sdr/rfsoc_sam/files/7460244/tcl_scripts.zip)
)
We are feeding directly the LMK clock of ### Hz into ADC tile directly, so configuration of LMX chips on the CLK104 card doesnot matter as per the screenshots attached.

**Comment by dnorthcote** [comment]
Hi Kaushal,

Firstly, just an update on the RFSoC SAM support for the ZCU216. Currently, we are not including this in our plan at the moment. The main reason for this decision is because the ZCU216 is currently not planned to have formal PYNQ support. See the following link: https://discuss.pynq.io/t/pynq-for-zcu216-rfsoc-board/3056

The reason your design may not be working could be due to the sample rate configuration set during the DynamicPLLConfig call for the [ADC](https://github.com/strath-sdr/rfsoc_sam/blob/597eb1f8dbcafba1b022ed7dcf2ca75c1c1e1f60/rfsoc_sam/hierarchies.py#L146) and [DAC](https://github.com/strath-sdr/rfsoc_sam/blob/597eb1f8dbcafba1b022ed7dcf2ca75c1c1e1f60/rfsoc_sam/hierarchies.py#L242). At the lines indicated, the sample rate is set to 4096Msps. You will need to set this to the sample rate configured in your IP Integrator design.

That's all I have for now, but can't necessarily spend more time on this. Happy to leave this issue open for anyone else with the same problem,

Thanks,
David.

**Comment by KAUSHAL1986** [comment]
I have commmented out dynamic configuration part for 409.6 MHz because I am configuring the clocks outside of rfsoc Sam project...
Thanks for the input. I understand you cannot spend time further on this.
Can you please list steps to port pynq onto zcu216 board...nothing useful I got at the link mentioned above...
Are tcl files of no use for debugging
With thanks and regards

**Comment by pedromiguelsantos** [comment]
I am also very interested in adding support for the ZCU216. I have ported and built PYNQ for the ZCU216 board using the agnostic prebuilt image and the bsp file (still unsure if it supports all the required devices).

On the IP side, the blocks should be compatible if we fix the clock, sample rates and tiling configuration right?

On the driver side, do we need Xilinx to add ZCU216 board support on the xrfclk, xrfdc and xsdfec? Is this the main problem @dnorthcote ?

@KAUSHAL1986 have you made any progress?

Regards,
Pedro

**Comment by KAUSHAL1986** [comment]
@pedromiguelsantos 
1. Following the pynq documentation for I have compiled an kernel which works fine with the PYNQ 2.6 file system .
2. I have modified the xrfclk drivers for zcu216 board partially as a quick fix and stuck with the xrfdc as you have noticed above.
3. I am planning to check xrfdc extensively first with C drivers as available at https://github.com/Xilinx/embeddedsw/tree/master/XilinxProcessorIPLib/drivers/rfdc
4. I have opened an issue @ https://github.com/Xilinx/ZCU111-PYNQ/issues/31 but nobody has responded as of now.
5. I will update if I am able to make xrfdc drivers work properly
